### ISSUE TITLE
[Bug] Reduce with max built-in having different behavior comparing to UDF

### ISSUE THREAD

**Issue by VoVAllen** [dmlc/dgl#761]
## 🐛 Bug
Node with no in-edge will be initialized as -inf, which is inconsistent with UDF.
Sometimes also fails CI test.

## To Reproduce

1.
1.
1.

## Environment

 - DGL Version (e.g., 1.0):
 - Backend Library & Version (e.g., PyTorch 0.4.1, MXNet/Gluon 1.3):
 - OS (e.g., Linux):
 - How you installed DGL (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version (if applicable):
 - GPU models and configuration (e.g. V100):
 - Any other relevant information:

**Comment by yzh119** [comment]
Please note this is also the case for `batched_graph.max_nodes`.

**Comment by jermainewang** [comment]
Could you paste the CI failure message here?

**Comment by VoVAllen** [comment]
http://ci.dgl.ai/blue/organizations/jenkins/DGL/detail/PR-750/3/pipeline

**Comment by yzh119** [comment]
@jermainewang I think it's not appropriate to return `-inf` for zero degree nodes/graphs. If user use the returned `-inf` feature to do classification, they would get a gradient with all `nan`.

```python
>>> import torch
>>> import torch.nn.functional as F
>>> W = torch.rand(3, 4, requires_grad=True)
>>> W
tensor([[0.3388, 0.3281, 0.8115, 0.8196],
        [0.0738, 0.7006, 0.4559, 0.7633],
        [0.0359, 0.5973, 0.2692, 0.8890]], requires_grad=True)
>>> x = torch.zeros(1, 3).fill_(-float('inf'))
>>> y = x @ W
>>> y
tensor([[-inf, -inf, -inf, -inf]], grad_fn=<MmBackward>)
>>> loss = F.cross_entropy(y, torch.LongTensor([1]))
>>> loss
tensor(nan, grad_fn=<NllLossBackward>)
>>> loss.backward(torch.ones(1))
>>> W.grad
tensor([[nan, nan, nan, nan],
        [nan, nan, nan, nan],
        [nan, nan, nan, nan]])
```
